test/contacts.js

```diff
@@ -78,172 +126,232 @@ describe('contacts', function() {
   })
 
   describe('getByEmail', function() {
-    it('should return a contact record based on the email', function() {
-      return hubspot.contacts
-        .getByEmail('testingapis@hubspot.com')
-        .then(data => {
-          expect(data).to.be.a('object')
-          expect(data.properties).to.be.a('object')
+    let email = 'testingapis@hubspot.com'
+    const getContactByEmailEndpoint = {
+      path: `/contacts/v1/contact/email/${email}/profile`,
+      response: { properties: {} },
+    }
+    fakeHubspotApi.setupServer({ getEndpoints: [getContactByEmailEndpoint] })
+
+    before(function() {
+      if (process.env.NOCK_OFF) {
+        return hubspot.contacts.get({ count: 3 }).then(data => {
+          email = emailsFromContacts(data.contacts)[0]
         })
+      }
+    })
+
+    it('should return a contact record based on the email', function() {
+      return hubspot.contacts.getByEmail(email).then(data => {
+        expect(data).to.be.a('object')
+        expect(data.properties).to.be.a('object')
+      })
     })
   })
 
   describe('getByEmailBatch', function() {
-    it('should return a contact record based on a array of emails', function() {
-      return hubspot.contacts
-        .getByEmailBatch([
-          'testingapis@hubspot.com',
-          'testingapisawesomeandstuff@hubspot.com',
-        ])
-        .then(data => {
-          expect(data).to.be.an('object')
+    let emails = [
+      'testingapis@hubspot.com',
+      'testingapisawesomeandstuff@hubspot.com',
+    ]
+    const getContactByEmailsEndpoint = {
+      path: '/contacts/v1/contact/emails/batch',
+      response: { properties: {} },
+      query: { email: emails },
+    }
+    fakeHubspotApi.setupServer({ getEndpoints: [getContactByEmailsEndpoint] })
+
+    before(function() {
+      if (process.env.NOCK_OFF) {
+        return hubspot.contacts.get({ count: 3 }).then(data => {
+          emails = emailsFromContacts(data.contacts)
         })
+      }
+    })
+
+    it('should return a contact record based on a array of emails', function() {
+      return hubspot.contacts.getByEmailBatch(emails).then(data => {
+        expect(data).to.be.an('object')
```

Comment:
I think it's wise to avoid testing too much about the return response since that logic lives with the HubSpot API but is there anything more meaningful we can assert for here? Perhaps one key/field that is relevant to a Contact?

Reply:
The response from HubSpot is an object with the keys set to the ids of the records. Because I won't know what these are beforehand, I don't think I should make any assertions beyond that it succeeds and I get back an object.